asf_public_discourse_home_decarbonisation/utils/text_processing_utils.py

```diff
@@ -0,0 +1,205 @@
+"""
+Utils for processing text data.
+"""
+
+import re
+from nltk.corpus import stopwords
+from nltk import ngrams
+import nltk
+
+nltk.download("stopwords")
+from gensim.parsing.preprocessing import STOPWORDS
+import re
+import string
+from nltk.stem.wordnet import WordNetLemmatizer
+from nltk.tag import pos_tag
+
+from nltk import pos_tag
+from nltk.corpus import wordnet
+
+nltk.download("punkt")
+nltk.download("averaged_perceptron_tagger")
+nltk.download("wordnet")
+import pandas as pd
+from collections import Counter
+
+
+def remove_urls(text: str) -> str:
+    """
+    Removes URLs from text.
+    Args:
+        text: a string, tipically one or multiple sentences long
+    Returns:
+        text without URLs
+    """
+    pattern = r"http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\\(\\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+"
+    cleaned_text = re.sub(pattern, " ", text)
+    return cleaned_text
+
+
+def replace_punctuation_with_space(text: str) -> str:
```

Comment:
imagine the following sentence, and not replacing with space:
"This is just an example.There was no space between the last sentence and this one, so example and there would be glued together"
-> "This is just an exampleThere was no space between the last sentence and this one so example and there would be glued together"

Reply:
That makes sense.
Do you know how it will handle contractions then? e.g. 'can't' will become 'can t' so you lose the meaning of the word. I'm not sure it's that important in the context of looking at keywords but it might become important if you start doing sentiment analysis.

Reply:
@crispy-wonton this processing is specific for this analysis. For future analyses it might not make sense to do lemmatisation, or other parts of the processing being applied here.